### ISSUE TITLE
Make arrow keys move to next and previous item with one press in Picker (dropdown list)

### ISSUE THREAD

**Issue by velara3** [adobe/spectrum-web-components#1187]
### Expected Behaviour
When a dropdown picker is focused I'd like to be able to select the next or previous items using the up and down arrows pressing them only one time. 

### Actual Behaviour
When the dropdown picker is focused and the menu options are not visible I press the up key or down key and the picker opens. Then on the second up or down key press the next or previous item is highlighted and then you must press the Enter key to select it (or click on the row). 

https://user-images.githubusercontent.com/1084127/107487650-3fa5b900-6b3b-11eb-962d-a635ff071c6b.mp4

### Reproduce Scenario (including but not limited to)

1. Go here:  https://opensource.adobe.com/spectrum-web-components/components/picker
2. Click on a picker so it has focus (and select something so the menu disappears again)
3. Press the up or down arrow key (the picker opens)
4. Press the up or down arrow key again (the next or previous row is highlighted)
5. Press the enter key to select the highlighted row

The behavior in the Picker is much better than the native dropdown but it is still horrible when you want to step through a long list of options and you can't use the arrow keys. 

For example, let's say you are in Photoshop and you want to change the blend mode of a layer. You have to open the dropdown and click on the row each time. The pop up menu items close each time you make a selection. 

Imagine how you would like to select each item in a list with the arrow keys and you can tell what I mean.  

### Platform
OSX using Chrome or Brave Browser

I don't think this is a Spectrum components problem but a problem with ALL dropdowns. The keyboard support is horrible at least on OSX (have not tried Windows in a while). 

Please consider it. Also, I would be satisfied if the behavior was enabled by flag or enabled by modifier key. For example, if you hold down ALT + Up arrow the next item is automatically selected (pop up doesn't open - simply the next item is selected).

**Comment by Westbrook** [comment]
Thanks for this issue. Very glad to hear we've surpassed the native `<select>` element in your view, and sorry to hear we've not gotten all the way to full coverage of your needs just yet. Further, I can tell you that we'd not be able to support this feature directly, though I really appreciate your giving us some alternatives that you'd be interested in instead.

From experience, I can tell you that even though your blend mode example really does point at the issue at hand, you likely don't want to commit a change to the `sp-picker` element's value in these contexts. This would make it difficult to disambiguate between the act of "choosing" a blend mode and "previewing" a blend mode. Without the clear distinction then "preview" actions could mistakenly get queued into an undo/redo stack causing other issues for your users that would be a much larger difficulty to address.

Before some changes to address the accessibility of this interface we would actually `focus` the individual menu items in the picker, which allowed for this sort of usage: https://webcomponents.dev/edit/uzQQ2npFGGehNjogMYKo/src/index.ts

```
        <sp-menu-item
            value="0"
            @focus=${(event) => this.current = event.target.value}
            @pointerenter=${(event) => this.current = event.target.value}
          >
            Immediately
          </sp-menu-item>
```

Herein the `focus`/`pointerenter` events would be able to manage the `current` value, while the `change` event on the `sp-menu` would mange the `committed` value. This is closer to the style of interaction management that I think you're looking for here. Walking through this now, it was a lot of extra code to synthesize the concept of an `input` event into the `sp-picker` interface. Surfacing that would actually ofter a good amount of flexibility for consumer while closing over a sizable amount of boilerplate that was required in the above example. We have a decent amount of work scheduled to go into the `sp-menu` family of elements coming up, and adding something like this at that time makes sense.

While a native `select` element does dispatch an `input` event, I can't seem to find a context where it does so without being directly in association with a `change` event: https://codepen.io/Westbrook/pen/ZEBpoMv I'll do a little more research there, to ensure we're not missing out of any functionality there, but I think that means we can _enhance_ the pattern as follows.

Let me know what you think about:

**Current Functionality**
`change` event is fired when the menu closes. This signifies that the `value` property is now the committed value.

**New Functionality**
`input` event is fired when an item becomes the `aria-activedescendant` of the menu element, e.g. keyboard focus is given to it and when a pointer enters an item. A subsequent `input` event would fire signifying the return to the value signified by `aria-activedescendant` or the  previously committed value when a pointer left an item. 

The opens one slightly surprising situation wherein the `aria-activedescendant` (`input` event with `value=1`) is set then an item is "hovered" (`input` event with `value=2`) and then the keyboard (`Enter` key) is used to make a selection, which in this case would commit `value=1` as that is the currently "active" item. It's possible that while addressing #1136 and normalizing with the `Picker` component in React Spectrum will remove this context, so hopefully it won't be an issue. For reference: the `aria-activedescendant` is updated to the "hovered" item therein in response to this interaction sequence.

**Comment by velara3** [comment]
> From experience, I can tell you that even though your blend mode example really does point at the issue at hand, you likely don't want to commit a change to the `sp-picker` element's value in these contexts. This would make it difficult to disambiguate between the act of "choosing" a blend mode and "previewing" a blend mode. Without the clear distinction then "preview" actions could mistakenly get queued into an undo/redo stack causing other issues for your users that would be a much larger difficulty to address.

Correct! If there was a `preview` event which sounds like the `[item]hovered` event then, using the Photoshop example, the application could listen for those `preview` events and show the user a preview of the blend mode. The Photoshop developers could use these hover events to commit the blend mode. However, I wouldn't want to implement this for each dropdown list in an application simply to navigate through each item in a list. 

Using the example it sounds like there can be two cases using keyboard interactions: 

- **Highlighting mode**: The Picker is focused and in an open state and the application listens for Picker `highlight` events to instruct the application. For example, a user wants to see what "multiply" blend mode looks like versus "normal" blend mode. He clicks on the Blend Mode picker, the pop up menu is opened and the current selected item is "normal", the user presses the down arrow to highlight the "multiply" blend mode and a `highlight` or `hovered` event is dispatched. The application responds to the event causing the design view to show a preview of the "multiply" blend mode. The user then presses the up arrow dispatching `hovered` events again and the "normal" blend mode is previewed in the design layer. The user decides he wants "multiply" blend mode so he presses the down arrow to highlight the "multiply" blend mode item and then presses enter to commit the highlighted item. The `change` event is fired and the new selected value is "multiply". There is only one action that shows up in the Undo and Redo history list. 

- **Direct Select mode**: The Picker is focused, the pop up is not visible (or is visible) and the up and down arrows directly select the next and previous items on each key press. The enter key or clicking on an item row are not needed to change the selected item. As an example, a user wants to see what all blend modes look like. He is currently at "normal" blend mode. He clicks on the Blend Mode picker, the pop up menu is opened and he clicks on the current selected item or presses enter again. This closes the pop up menu but keeps the Picker focused. The user then presses the down arrow to select the "multiply" blend mode. The `change` event is dispatched. The application responds to the `change` event causing the design view to use the "multiply" blend mode. The user then presses the down arrow again dispatching another `change` event and the next blend mode is selected in the design layer. There are multiple actions that show up in the Undo and Redo history list for each change. 

If the **Preview mode** already exists but under a different event names then theoretically you could provide the user with a way to preview each blend modes effect without having to open and close the Picker each time. That would solve the use case with the provided events. To view the next 3 items instead of click, down arrow, enter, down arrow, enter, down arrow, enter it would be focus (click), down arrow, down arrow, down arrow. This desired behavior could be accomplished with the existing or proposed events. 

If the **Direct Select mode** is at sometime supported or supported with modifier keys or with an option, the Photoshop developers could enable it via an option without having to change any of their code. If this option is enabled the user skips the opening and closing of the pop up menu and changes to the next or previous item in the Picker on up or down arrow key press. To Photoshop it's the same as before. Photoshop is receiving `change` events. It does not know the user is skipping the open and close of the pop up menu. 

In regards to the ARIA support I don't know enough to provide feedback here. I'm guessing you could skip the some of the events and go directly to the ARIA related Select control `change` event. 

If the developer has to set a flag that enables Direct Select mode that would enhance things for me. If the user can hold down a modifier key as they use the arrow keys to `select` or `change` rather than `highlight`, that would enhance things for me. 

**New Functionality**
As an example, using blend modes example. 

_First Case:_ 
Developer has enabled Direct Select behavior via attribute in the application. The user, viewing the application and the Blend Mode Picker is focused. He has focused it by tabbing to it. The pop up menu is not open. The first item, "Normal" is selected. 
- The user then presses the down arrow to select the "multiply" blend mode. The pop up menu is not opened. 
- The `change` event is dispatched and "multiply" item is selected
- The user presses the down arrow again dispatching another `change` event and the next blend mode is selected

_Second case:_ 
Developer has enabled Direct Select behavior via attribute in the application. The BlendMode Picker does not have focus. The user clicks on it and the pop up menu opens, is visible and focused. The first item, "Normal" is selected. 
- The user then presses the down arrow. The next row item is selected. The pop up menu remains open. 
- The `change` event is dispatched
- The user presses the down arrow again dispatching another `change` event and the next item is selected. The pop up menu remains open until the user tabs away, clicks outside, or presses the enter key. 

BTW I think the **Direct Select mode** is the way drop downs behave by default in Windows and Linux platforms. I think it is only different on OSX. Or maybe it's the way I've wanted it to work. I don't have access to a Windows PC at the moment to verify.

**Comment by Westbrook** [comment]
@velara3 We're adding support for selecting the "next/previous" item with the right/left arrow keys in https://github.com/adobe/spectrum-web-components/pull/1220 which may solve much of your original request.

Hopefully we can get into some of this extended context soon, as well.

**Comment by Westbrook** [comment]
Closed by #2700, which focuses in on the specifics of an "input mode".
